### PR TITLE
lesson 13: finish lesson

### PR DESCRIPTION
**Pull request by dlincheuski**

### CODE REVIEW

File: lesson 13.playground/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,135 @@
+// с оздать структуру с данными Имя, Фамилия, Год рождения, средний балл оценок
+struct Student {
+    var firstName: String
+    var sername: String
```
Comment (nograsby) [inline lesson 13.playground/Contents.swift]: surname

File: lesson 13.playground/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    var yearBirthday: Int
+    var equilBall: Double
```
Comment (nograsby) [inline lesson 13.playground/Contents.swift]: averageMark

File: lesson 13.playground/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,135 @@
+// с оздать структуру с данными Имя, Фамилия, Год рождения, средний балл оценок
+struct Student {
+    var firstName: String
+    var sername: String
+    var yearBirthday: Int
```
Comment (nograsby) [inline lesson 13.playground/Contents.swift]: nameUserComputerClub
computerClubUserName

File: lesson 13.playground/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+// сделать несколько экземпляров и заполнить их  данными
+var student1 = Student(firstName: "Lex", sername: "Fox", yearBirthday: 1994, equilBall: 7.8)
+var student2 = Student(firstName: "Pol", sername: "MgGill", yearBirthday: 1995, equilBall: 6.4)
+var student3 = Student(firstName: "Piter", sername: "MgGill", yearBirthday: 1994, equilBall: 8.9)
+var student4 = Student(firstName: "Yan", sername: "Dzi", yearBirthday: 1996, equilBall: 5.4)
+
+// положить экземпляры в массив
+var studentBookStructs = [student1, student2, student3, student4]
```
Comment (Nograsby) [inline lesson 13.playground/Contents.swift:16]: studentBooks

File: lesson 13.playground/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+// создать функцию, которая будет принимать массив и возвращать порядковый номер и данные студентов
+func returnStudentInfo (array : [Student]) {
```
Comment (nograsby) [inline lesson 13.playground/Contents.swift]: printSt..

Comment (nograsby) [reply lesson 13.playground/Contents.swift]: students:

File: lesson 13.playground/Contents.swift [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+returnStudentInfo(array: studentBookStructs)
+
+// искользуя sorted отсортировать массив по среднему баллу по убыванию и распечатать
+print(studentBookStructs.sorted { (a, b) -> Bool in
+    return a.equilBall < b.equilBall
+})
+
+// -//- по фамилии, если фамилии одинаковы то сортировка должна быть по имени и распечатать
+func studentBookStructsSortedSername (array: [Student]) {
```
Comment (nograsby) [inline lesson 13.playground/Contents.swift]: sortStudentsBySurname(_ students:

File: lesson 13.playground/Contents.swift [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+// сделать несколько экземпляров и заполнить их  данными
+var studentClass1 = StudentClass(firstName: "Lex", sername: "Fox", yearBirthday: 1994, equilBall: 7.8)
+var studentClass2 = StudentClass(firstName: "Pol", sername: "MgGill", yearBirthday: 1995, equilBall: 6.4)
+var studentClass3 = StudentClass(firstName: "Piter", sername: "MgGill", yearBirthday: 1994, equilBall: 8.9)
+var studentClass4 = StudentClass(firstName: "Yan", sername: "Dzi", yearBirthday: 1996, equilBall: 5.4)
+
+// положить экземпляры в массив
+var studentBookClass = [studentClass1, studentClass2, studentClass3, studentClass4]
```
Comment (nograsby) [inline lesson 13.playground/Contents.swift]: es

File: lesson 13.playground/Contents.swift [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+// создать функцию, которая будет принимать массив и возвращать порядковый номер и данные студентов
+func returnStudentInfo (array : [StudentClass]) {
```
Comment (nograsby) [inline lesson 13.playground/Contents.swift]: print

File: lesson 13.playground/Contents.swift [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+    print("Действие функции по классу по информации из журнала")
+    for i in 0 ..< (array.count) {
+        print("\(i+1). Имя - \(array[i].firstName), Фамилия - \(array[i].sername), Год рождкния - \(array[i].yearBirthday)г., средний балл - \(array[i].equilBall)  ")
+    }
+}
+returnStudentInfo(array: studentBookClass)
+
+// искользуя sorted отсортировать массив по среднему баллу по убыванию и распечатать
+print(studentBookClass.sorted { (a, b) -> Bool in
```
Comment (Nograsby) [inline lesson 13.playground/Contents.swift:100]: оформить в функцию

File: lesson 13.playground/Contents.swift [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+            if (studentBookClassSorted[j].sername == studentBookClassSorted[element].sername) && (studentBookClassSorted[j].firstName > studentBookClassSorted[element].firstName) {
+                let temp = studentBookClassSorted[j]
+                studentBookClassSorted[j] = studentBookClassSorted[element]
+                studentBookClassSorted[element] = temp
+            }
+        }
+    }
+    
+    for i in 0 ..< studentBookClassSorted.count {
+        print("\(i+1). Имя - \(studentBookClassSorted[i].firstName), Фамилия - \(studentBookClassSorted[i].sername), Год рождкния - \(studentBookClassSorted[i].yearBirthday)г., средний балл - \(studentBookClassSorted[i].equilBall)  ")
```
Comment (nograsby) [inline lesson 13.playground/Contents.swift]: i + 1

File: lesson 13.playground/Contents.swift [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+var student1 = Student(firstName: "Lex", surname: "Fox", birthdayYear: 1994, averageMark: 7.8)
+var student2 = Student(firstName: "Pol", surname: "MgGill", birthdayYear: 1995, averageMark: 6.4)
+var student3 = Student(firstName: "Piter", surname: "MgGill", birthdayYear: 1994, averageMark: 8.9)
+var student4 = Student(firstName: "Yan", surname: "Dzi", birthdayYear: 1996, averageMark: 5.4)
+
+// положить экземпляры в массив
+var studentBooksStruct = [student1, student2, student3, student4]
+
+// создать функцию, которая будет принимать массив и возвращать порядковый номер и данные студентов
+func printStudentInfo (students : [Student]) {
```
Comment (nograsby) [inline lesson 13.playground/Contents.swift]: func printStudentsInfo(_ students : [Student]) {

File: lesson 13.playground/Contents.swift [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+// сделать несколько экземпляров и заполнить их  данными
+var student1 = Student(firstName: "Lex", surname: "Fox", birthdayYear: 1994, averageMark: 7.8)
+var student2 = Student(firstName: "Pol", surname: "MgGill", birthdayYear: 1995, averageMark: 6.4)
+var student3 = Student(firstName: "Piter", surname: "MgGill", birthdayYear: 1994, averageMark: 8.9)
+var student4 = Student(firstName: "Yan", surname: "Dzi", birthdayYear: 1996, averageMark: 5.4)
+
+// положить экземпляры в массив
+var studentBooksStruct = [student1, student2, student3, student4]
```
Comment (nograsby) [inline lesson 13.playground/Contents.swift]: studentsStructures

**Review by nograsby — APPROVED** [review]

**Pull request merged**